code/controllers/master.dm

```diff
@@ -675,3 +695,79 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
 			. = "<font color='#eb4034'>[cpu_var]</font>"
 		if(100 to INFINITY) // >100 = bold red
 			. = "<font color='#eb4034'><b>[cpu_var]</b></font>"
+
+// Updates SS loading stuff on the lobby
+/datum/controller/master/proc/update_ss_loadingscreen(current_ss_id)
+	// We are done, clear it
+	if(!length(ss_in_init_order))
+		GLOB.title_splash.maptext = null
+		return
+
+	var/list/columns = list()
+	columns += list(list()) // Init our first column
+
+	var/spacer = "        " // 8 characters width space
+	// You can comfortably fit 33 lines of text on the lobby screen, so we have to do column memes to make the text work
+	var/max_per_col = 33
+
+
+	var/i = 0 // We dont iterate the below loop on i because we only care about initters
+	for(var/datum/controller/subsystem/SS in ss_in_init_order)
+		if(SS.flags & SS_NO_INIT)
+			continue
+
+		i++
+
+		var/list/target_col = columns[length(columns)] // Get most recent once
+
+		if(ceil(i / (max_per_col * length(columns))) > 1)
```

Comment:
SDMM thinks `ceil` doesnt exist.

Reply:
It's not exactly a new proc, did they just forget to add it?

Reply:
Turns out its a 515 thing

Reply:
Oh, I assumed it was like `round` and had been here forever

Reply:
![image](https://github.com/ParadiseSS13/Paradise/assets/25063394/9f26d5bc-98a8-45ee-8b17-80e6367d6173)